src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -62,29 +62,35 @@ public class CreateAlertTool implements Tool {
     private String description = DEFAULT_DESCRIPTION;
 
     private final Client client;
+    @Getter
     private final String modelId;
-    private final String TOOL_PROMPT_TEMPLATE;
+    @Getter
+    private final String toolPrompt;
 
     private static final String MODEL_ID = "model_id";
     private static final String PROMPT_FILE_PATH = "CreateAlertDefaultPrompt.json";
     private static final String DEFAULT_QUESTION = "Create an alert as your recommendation based on the context";
     private static final Map<String, String> promptDict = ToolHelper.loadDefaultPromptDictFromFile(CreateAlertTool.class, PROMPT_FILE_PATH);
 
-    public CreateAlertTool(Client client, String modelId, String modelType) {
+    public CreateAlertTool(Client client, String modelId, String modelType, String prompt) {
         this.client = client;
         this.modelId = modelId;
-        if (!promptDict.containsKey(modelType)) {
-            throw new IllegalArgumentException(
-                LoggerMessageFormat
-                    .format(
-                        null,
-                        "Failed to find the right prompt for modelType: {}, this tool supports prompts for these models: [{}]",
-                        modelType,
-                        String.join(",", promptDict.keySet())
-                    )
-            );
+        if (prompt.isEmpty()) {
+            if (!promptDict.containsKey(modelType)) {
```

Comment:
We can follow the behavior of PPL Tool. If modelType is an empty string `""`, then take its default value.

Reply:
FYI: https://github.com/opensearch-project/skills/blob/d0c42a8df6b43794f25f65e70f013695173582cd/src/main/java/org/opensearch/agent/tools/PPLTool.java#L136